### PR TITLE
[Proxy] Add proxy additional servlet plugin framework

### PR DESCRIPTION
**Pull request by gaoran10**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/plugin/servlet/ProxyAdditionalServletDefinition.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.proxy.server.plugin.servlet;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Metadata information about a proxy protocol handler.
```
Comment (codelipenghui) [inline pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/plugin/servlet/ProxyAdditionalServletDefinition.java:25]: Not a protocol handler, we named additional servlet right?

Comment (gaoran10) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/plugin/servlet/ProxyAdditionalServletDefinition.java]: Yes, I'll fix it.

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/plugin/servlet/ProxyAdditionalServletDefinition.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+@Data
+@NoArgsConstructor
+public class ProxyAdditionalServletDefinition {
+    /**
+     * The name of the broker interceptor.
```
[the remark this answers is not in the document]
Comment (gaoran10) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/plugin/servlet/ProxyAdditionalServletDefinition.java]: OK

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java [hunk 3]
```diff
@@ -159,6 +162,12 @@
     )
     private Optional<Integer> webServicePortTls = Optional.empty();
 
+    @FieldContext(
+            category = CATEGORY_SERVER,
+            doc = "The directory where nar Extraction of offloaders happens"
+    )
+    private String narExtractionDirectory = NarClassLoader.DEFAULT_NAR_EXTRACTION_DIR;
```
[the remark this answers is not in the document]
Comment (gaoran10) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java]: It could be removed.

**Review by codelipenghui — APPROVED** [review]

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gaoran10) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java]: This config is useful, I recover it and fix the description.

**Pull request merged**